esp-hal/src/rmt.rs

```diff
@@ -253,14 +253,34 @@ where
             return Err(Error::UnreachableTargetFrequency);
         }
 
-        let div = (src_clock / frequency) - 1;
+        let quotient = src_clock / frequency;
+        let div = quotient - 1;
 
         if div > u8::MAX as u32 {
             return Err(Error::UnreachableTargetFrequency);
         }
 
-        self::chip_specific::configure_clock(div);
+        // Calculate the greatest common divisor of a and b
+        fn gcd(mut a: u32, mut b: u32) -> u32 {
+            while b != 0 {
+                (a, b) = (b, (a % b))
+            }
+            a
+        }
 
+        let mut div_a = 0;
+        let mut div_b = 0;
+        let remainder = src_clock.raw() % frequency.raw();
+        if remainder > 0 {
+            let gcd = gcd(frequency.raw(), remainder);
+            let numerator = remainder / gcd;
+            let denominator = frequency.raw() / gcd;
+            if numerator < (1 << 5) && denominator < (1 << 5) {
+                div_a = denominator;
+                div_b = numerator;
+            }
+        }
+        self::chip_specific::configure_clock(div, div_a, div_b);
```

Comment:
There's a fractional clock solver in the LCD_CAM driver https://github.com/esp-rs/esp-hal/blob/e1c27f1b22e4300ecd4d7bea970f4f72d3ba632e/esp-hal/src/lcd_cam/mod.rs#L270-L359 .
Would be cool if you could reuse it. (Or it can be left for someone else to look at in a different PR)

Reply:
I checked the solver and noticed there's a little bit different on the register definition of `div_num`, which may cause we cannot easily reuse the code.

The farey sequence code can be reused as a solution when the denominator after rounding exceeds the maximum limit.

Reply:
`div_num` differences can be simply solved by overflowing adding `1`.